core/src/main/java/org/elasticsearch/index/query/ExistsQueryParser.java

```diff
@@ -57,6 +57,8 @@ public ExistsQueryBuilder fromXContent(QueryParseContext parseContext) throws IO
                 } else {
                     throw new ParsingException(parser.getTokenLocation(), "[exists] query does not support [" + currentFieldName + "]");
                 }
+            } else {
+                throw new ParsingException(parser.getTokenLocation(), "[exists] unknown token [" + token + "] after [" + currentFieldName + "]");
```

Comment:
You switched to using the query builder NAME in other places - why not here as well?